apps/portal/src/content/docs/components/button-options.mdx

```diff
@@ -0,0 +1,226 @@
+        ]}
+        handleOptionChange={(val) => alert(\`Selected: \${val}\`)}
+        theme="primary"
+        loading={true}
+      >
+        Loading....
+      </ButtonWithOptions>
+    </div>
+    <div className="mb-4 text-white">
+      <div className="text-lg font-bold">Error Button (Default Size)</div>
+      <ButtonWithOptions
+        id="error-button-default"
+        handleButtonClick={() => alert('Error Button clicked')}
+        options={[
+          { value: 'option1', label: 'Option 1', description: 'Description for option 1' },
+          { value: 'option2', label: 'Option 2', description: 'Description for option 2' },
+        ]}
+        handleOptionChange={(val) => alert(\`Selected: \${val}\`)}
+        theme="error"
+      >
+        Error Button
+      </ButtonWithOptions>
+    </div>
+    <div className="mb-4 text-white">
+      <div className="text-lg font-bold">Success Button (Default Size)</div>
+      <ButtonWithOptions
+        id="success-button-default"
+        handleButtonClick={() => alert('Success Button clicked')}
+        options={[
+          { value: 'option1', label: 'Option 1', description: 'Description for option 1' },
+          { value: 'option2', label: 'Option 2', description: 'Description for option 2' },
+        ]}
+        handleOptionChange={(val) => alert(\`Selected: \${val}\`)}
+        theme="success"
+      >
+        Success Button
+      </ButtonWithOptions>
+    </div>
+  </div>`}
+/>
+
+## Usage
+
+```typescript jsx
+import { ButtonWithOptions } from '@harnessio/ui/components'
+
+return (
+  <div className="space-y-4">
+    <div className="mb-4 text-white">
+      <div className="text-lg font-bold">Primary Button (Default Size)</div>
+      <ButtonWithOptions
+        id="primary-button-default"
+        handleButtonClick={() => alert('Primary Button clicked')}
+        options={[
+          { value: 'option1', label: 'Option 1', description: 'Description for option 1' },
+          { value: 'option2', label: 'Option 2', description: 'Description for option 2' },
+        ]}
+        handleOptionChange={(val) => alert(\`Selected: \${val}\`)}
+        theme="primary"
+      >
+        Primary Button
+      </ButtonWithOptions>
+    </div>
+  </div>
+)
+```
+
+## Anatomy
+
+All parts of the ButtonWithOptions component can be imported and composed as required.
+
+```typescript jsx
+<ButtonWithOptions
+  id="example-button-with-options"
+  handleButtonClick={() => alert('Button clicked')}
+  options={[
+    { value: 'option1', label: 'Option 1', description: 'Description for option 1' },
+    { value: 'option2', label: 'Option 2', description: 'Description for option 2' },
+  ]}
+  handleOptionChange={(val) => alert(\`Selected: \${val}\`)}
+>
+  Button Text
+</ButtonWithOptions>
+```
+
+## API Reference
+
+### ButtonWithOptions
+
+The `ButtonWithOptions` component serves as the main container for the button and its dropdown options.
```

Comment:
We can drop this too since it's covered elsewhere and isn't really true anyway (to the user of the component, it's not a container for something, it's its own thing).